lib/python/ensembl/brc4/runnable/compare_fasta.py

```diff
@@ -212,127 +174,129 @@ def compare_seqs(self, seq1, seq2):
-                    only1_checksum.pop(name1)
-            for seq_2, name2 in list(only2.items()):
-                name2 = str(name2)
-                if core == name2:
-                    del only2[seq_2]
-                    del only2_checksum[name2]
-
-        if len(seq1) == len(seq2) and len(common) == len(seq1):
-            if len(only1) == 0 and len(only2) == 0:
-                value = "identical"
-        else:
+        if only1 or only2:
             value = "mismatch"
+
         # Gathering the organellar sequences
         report = self.param_required("report")
         report_parser = SeqregionParser()
-        report_seq = report_parser.get_report_regions(report)
-        report = self.add_report_to_map(common, report_seq)
+        report_seq = report_parser.get_report_regions(report, accession)
         map_dna_path = self.param_required("seq_regions")
-        data = self.get_json(map_dna_path)
-        org, location = self.organellar(report_seq, data)
-
-        # check ids for all the matched and mis-matched sequences
-        All_id = {}
-        for insdc, coredb_name in report_seq.items():
-            All_id[insdc] = coredb_name["name"]
-        for insdc1, core2 in common.items():
-            if insdc1 in All_id and core2 in All_id.values():
-                pass
-            else:
-                comp.append("%s do not match in common" % insdc)
+        seq_data = self.get_json(map_dna_path)
+        org_loc = self.organellar_assembly(report_seq, seq_data)
+        INSDC_assembly_level, core_assembly_level = self.assembly_level(
+            report_seq, seq_data
+        )
 
-        only1_id = [name for seq, name in only1.items()]
-        only2_id = [name for seq, name in only2.items()]
+        comp.append(
+            "Assembly level: %s vs %s" % (INSDC_assembly_level, core_assembly_level)
+        )
 
-        for i in only1_id:
-            i = str(i)
-            for insdc, core in All_id.items():
-                if i == insdc and i != core:
-                    comp.append("%s and %s match by id" %
-                                (i, core))
-                else:
-                    pass
+        names_length = {}
+        # sequences which have extra N at the end
+        if only1 and only2:
+            for seq1, name1 in only1.items():
+                len1 = len(seq1)
+                seq1_N = seq1.count("N")
+                for seq2, name2 in only2.items():
+                    len2 = len(seq2)
+                    seq2_N = seq2.count("N")
+                    sequence_2 = seq2[:len1]
+                    if sequence_2 == seq1:
+                        ignored_seq = seq2[len1:]
+                        N = ignored_seq.count("N")
+                        if len(ignored_seq) == N:
+                            comp.append(
+                                f"Please check extra Ns added in core in  %s and %s"
+                                % (name1, name2)
+                            )
```

Comment:
```suggestion
                            comp.append(f"Please check extra Ns added in core in {name1} and {name2}")
```
Mixed used of `%` and f-string.